specs/LaunchingRegisteredProtocols.md

```diff
@@ -0,0 +1,283 @@
+  /// handle this event by suppressing the popup dialog that gives the user
+  /// the option to allow the protocol launch. The host also has the option to
+  /// programatically cancel the protocol launch.
+  /// The host also is given the opportunity to disable the always open checkbox 
+  /// which also revokes previous permissions given to this protocol/origin.
+  /// The `NavigationStarting`, `NavigationCompleted, `SourceChanged`,
+  /// `ContentLoading`, and `HistoryChanged` events will not be raised, regardless
+  /// of whether the `Cancel` or `Handled` property is set to `TRUE` or
+  /// `FALSE`. This behavior holds true for the frame navigation events as
+  /// well.
+  HRESULT add_LaunchingExternalProtocol(
+      [in] ICoreWebView2LaunchingExternalProtocolEventHandler* eventHandler,
+      [out] EventRegistrationToken* token);
+
+  /// Remove an event handler previously added with
+  /// `add_LaunchingExternalProtocol`.
+  HRESULT remove_LaunchingExternalProtocol(
+      [in] EventRegistrationToken token);
+}
+
+// This is the ICoreWebView2Frame3 interface.
+[uuid(fe1d3718-fe8d-48ab-8594-9e3fff6755ac), object, pointer_default(unique)] 
+interface ICoreWebView2Frame3 : IUnknown {
+  /// Add an event handler for the `LaunchingExternalProtocol` event.
+  /// A frame will raise a `LaunchingExternalProtocol` event when any non-main
+  /// frames attempt to launch an external protocol. This event aligns with the 
+  /// event raised on the `CoreWebView2` interface. 
+
+  HRESULT add_LaunchingExternalProtocol(
+      [in] ICoreWebView2FrameLaunchingExternalProtocolEventHandler* eventHandler,
+      [out] EventRegistrationToken* token);
+
+  /// Remove an event handler previously added with `add_LaunchingExternalProtocol`.
+  HRESULT remove_LaunchingExternalProtocol(
+      [in] EventRegistrationToken token);
+}
+
+/// Receives `LaunchingExternalProtocol` events for main frame.
+[uuid(e5fea648-79c9-47aa-8314-f471fe627649), object, pointer_default(unique)] 
+interface ICoreWebView2LaunchingExternalProtocolEventHandler: IUnknown {
+  /// Provides the event args for the corresponding event.
+  HRESULT Invoke(
+      [in] ICoreWebView2* sender,
+      [in] ICoreWebView2LaunchingExternalProtocolEventArgs* args);
+}
+
+/// Receives `LaunchingExternalProtocol` events for non-main frame.
+[uuid(a898c12c-f949-474c-913b-428770cfc177), object, pointer_default(unique)] 
+interface ICoreWebView2FrameLaunchingExternalProtocolEventHandler: IUnknown {
+  /// Provides the event args for the corresponding event.
+  HRESULT Invoke(
+      [in] ICoreWebView2Frame* sender,
+      [in] ICoreWebView2LaunchingExternalProtocolEventArgs* args);
+}
+
+/// Event args for `LaunchingExternalProtocol` event.
+[uuid(fc43b557-9713-4a67-af8d-a76ef3a206e8), object, pointer_default(unique)] 
+interface ICoreWebView2LaunchingExternalProtocolEventArgs: IUnknown {
+  /// The uri of the requested protocol.
+
+  [propget] HRESULT Uri([out, retval] LPWSTR* uri);
```

Comment:
Should we also have the URI that is doing the launching (maybe call it ReferrerUri) in addition to the URI to be launched? In the case of replacing the functionality of the default UI this would be necessary to implement their own 'always allow' sort of feature. 

Although if you have the event on the CoreWebView2/Frame perhaps the end dev can check the URI of the document from the CoreWebView2/Frame event source and so its not necessary to duplicate it here. Unless you move this to one event on the CoreWebView2 for all child frames in which case we would need this property.